litellm/proxy/proxy_server.py

```diff
@@ -1358,97 +1355,14 @@ async def _run_background_health_check():
             await asyncio.sleep(health_check_interval)
 
 
-class ProxyConfig:
+class ProxyConfig(BaseProxyConfig):
     """
     Abstraction class on top of config loading/updating logic. Gives us one place to control all config updating logic.
     """
 
     def __init__(self) -> None:
         pass
 
-    def is_yaml(self, config_file_path: str) -> bool:
```

Comment:
@ishaan-jaff 

instead of such a large refactor, why not 

- load the yaml 
- run a function to parse all values in the yaml and load in any `os.environ/` 
- set the updated dictionary as the config value before it enters any of our other checks 
?